### PR TITLE
Implement MOLECULE_BASE_FILE and accept list input to molecule_from_yaml filter

### LINKED ISSUE #1423: base-config not represented in ansible provisioner `molecule_yml`
# Issue Type

- Bug report

# Molecule and Ansible details

```
$ ansible --version                                                                                                                      
ansible 2.6.2
```

```
$ molecule --version
molecule, version 2.16.0
```

Molecule installation method (one of):

- pip

Ansible installation method (one of):

- pip

# Desired Behavior

I have a monolith repo with many roles (50+) so there is value to me to have a shared `molecule.yml`.

The directory structure looks like
```
molecule/
  resources/
    molecule.yml <-- base config
roles/
  A/
    molecule/
      default/
        molecule.yml
  B/
    molecule/
      default/
        molecule.yml <-- role config
```

The `base-config` consists of
```
---
dependency:
  name: galaxy
driver:
  name: lxd
platforms:
  - name: instance-xenial
    ...
```
and the `role-config` files are empty.

I expect that running `molecule --base-config ../../molecule/resources/molecule.yml destroy` would merge the two files and preserve the `platforms` configuration.

# Actual Behaviour (Bug report only)

`platforms` is not preserved, and running `molecule --base-config ../../molecule/resources/molecule.yml destroy` yields:
```
$ MOLECULE_DEBUG=True molecule -c ../../molecule/resources/molecule.yml destroy
--> Validating schema /home/me/src/molecule/ansible-playbooks/roles/docker/molecule/default/molecule.yml.
Validation completed successfully.

--> Test matrix
    
└── default
    └── destroy
    
--> Scenario: 'default'
--> Action: 'destroy'
    
    PLAY [Destroy] *****************************************************************
    
    TASK [Destroy molecule instance(s)] ********************************************
    fatal: [localhost]: FAILED! => {"msg": "'dict object' has no attribute 'platforms'"}
    
    PLAY RECAP *********************************************************************
    localhost                  : ok=0    changed=0    unreachable=0    failed=1   
    
    
ERROR:
```

Maybe this is a dup of #328 but I do not expect an empty `role-config` file to overwrite the `base-config`.

### PR DESCRIPTION
**Pull request by uSpike**
Fixes #1423

### CODE REVIEW

File: molecule/config.py [hunk 1]
```diff
@@ -185,6 +293,7 @@ def env(self):
         return {
             'MOLECULE_DEBUG': str(self.debug),
             'MOLECULE_FILE': self.molecule_file,
+            'MOLECULE_BASE_FILE': self.molecule_base_file,
```
Comment (retr0h) [inline molecule/config.py]: Needs added to tests.

https://github.com/metacloud/molecule/blob/master/test/unit/test_config.py#L286

**Comment by retr0h** [comment]
I'm not a huge fan of this feature in general, b/c it is over complicating things.  However, it is fixing a bug which I totally overlooked.

The more I work on issues around the base config, the less and less of a fan I become.  It seems overly complicated for very little benefit.  But, hey, we are where we are and your PR seems to address it.

**Comment by retr0h** [comment]
Also, can you please run `tox -e format` when finished.

**Comment by themr0c** [comment]
close/repopren to trigger travis build.

**Comment by uSpike** [comment]
I've rebased this and fixed linting errors.  Failing tests were not related to this PR, AFAIK.

**Comment by themr0c** [comment]
Thank you so much to bring this PR back to life!

Since the handover to Red Hat, all commits need to be signed with the `--signoff` option, that's the meaning of the DCO failure. Would you mind amending your commits to follow the new rule? I would be very grateful.

**Comment by uSpike** [comment]
I think I got the DCO right now.

**Comment by uSpike** [comment]
Tests are failing because I missed some updates, fixing...

**Comment by uSpike** [comment]
There's a number of failures that don't *seem* related, but I'm not sure.  Anyone have ideas what's wrong?

**Comment by decentral1se** [comment]
Lives on over at https://github.com/ansible/molecule/pull/2193. Let's follow up there.

**Pull request closed without merging**